os/fs/smartfs/smartfs_utils.c

```diff
@@ -1579,6 +1579,10 @@ int smartfs_deleteentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *en
 			if (ENTRY_VALID(direntry)) {
 				/* Count this entry */
 				count++;
+				if (count == 2) {
```

Comment:
good! =D I like it. but below is better.
```
if (ENTRY_VALID(direntry)) {
     /* If there's already valid entry, break it */
     if (count == 1) {
           break;
     }
     count++;
}
```

Reply:
This way, when the loop exits with count = 1
We don't know if the loop ended naturally
OR it was due to the break statement.

Reply:
ah yes. you're right.